src/background/tokenBackground.ts

```diff
@@ -101,6 +108,61 @@ export default class TokenBackground {
     chrome.runtime.sendMessage({ type: MESSAGE_TYPE.QRC_TOKEN_BALANCES_RETURN, tokens: this.tokens });
   }
 
+  private getQRCTokenDetails = async (contractAddress: string) => {
+    let msg;
+    /*
+    * Further contract address validation - if the addr provided does not have name,
+    * symbol, and decimals fields, it will throw an error as it is not a valid
+    * qrc20TokenContractAddr
+    */
+    try {
+      const nameRes = await this.bg.rpc.callContract(
```

Comment:
nit: `let res = await ...`
and use the same `res` for all the rpc calls so you dont have to allocate 3 different `res` vars